The Mass Driver Update

DeityLink opened the pull request:
![massdriverupdate](https://cloud.githubusercontent.com/assets/7573912/7220205/6a19cdae-e6bf-11e4-9d1e-e112f314e4fa.png)
- Mass Drivers and their buttons are now buildable and deconstructable.
- Mass Drivers can now be emagged, quintupling the velocity at which they throw things
- Completely revamped the "Pod Doors and Mass Drivers Control" computer, it can now store multiple ID_Tags, and even fire all of their respective drivers simultaneously!
- Made a Deathsquad version of this computer, that can operate the Mass Drivers in centcomm and send the mechs to various ZLevels.
- Buttons can trigger a Mass Driver at a maximal range of 7, Computers cover the entire Z-Level
- Moved the piano back inside the bar, as was often requested
- Added a few Entertainment Monitors in a handful of fitting place throughout the station
- Made a new sprite for the Tcomms Server computer (was previously using the sprite of the PDA Messages monitoring computer)

Might find a way to build pod doors in the futur, now that we have an easy way to toggle them.

clusterfack commented:
Dem lines changed

DeityLink commented:
I forgot to clean the maps, >_>

clusterfack commented:
This isn't a nitpick about your pull but we need to really to offload mounted frames to a secondary menu just for them alone soon because they are becoming quite a big list.

DeityLink commented:
aye, we'll have to find something ergonomic.

ComicIronic commented on code/game/machinery/mass_driver.dm:
```diff
@@ -9,34 +8,198 @@
+		if(3) // Wired
+			if(istype(W, /obj/item/weapon/wirecutters))
+				user << "You begin to remove the wiring from \the [src]."
+				if(do_after(user, 10))
```
Busy check

ComicIronic commented on code/game/machinery/mass_driver.dm:
```diff
@@ -9,34 +8,198 @@
+				var/obj/item/stack/rods/R=W
+				user << "You begin to complete \the [src]..."
+				playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+				if(do_after(user, 20) && (R.amount >= 2))
```
Busy check

clusterfack commented:
The airlock frames actually already have a secondary menu, but I do bemoan the lack of a back button.

clusterfack commented on code/game/machinery/mass_driver.dm:
```diff
@@ -9,34 +8,198 @@
+					user << "The welding tool must be on to complete this task."
+					return 1
+				playsound(get_turf(src), 'sound/items/Welder.ogg', 50, 1)
+				usr << "You begin to cut the frame apart..."
```
user

Exxion commented:
You thinking what I'm thinking?


![image](https://cloud.githubusercontent.com/assets/9827785/7221509/7f558d38-e6bb-11e4-9f94-833197a47f0a.png)

Manaony commented:
Instead of making another use for the Emag why not just having them hackable?

DeityLink commented:
That'd mean adding wires and some wire datums. Maybe later.

DeityLink commented:
* Added coat racks to detective offices on all three main stations.
* Added piano stools.
* The Control Computers' timers can now be set to loop individually. The ride never ends!
* if you increase a Mass Driver's power, it'll now also consume more power at each launch.

ComicIronic commented:
Muh atomic PRs

DeityLink commented:
It's the "Mass Driver Update" with in small letters in the corner "and also all the small little thing I wanted to do if I ever had the chance to make a map update which I usually don't since there's always a pending Map-related PR and also before we enter the Feature Freeze"... >_>

d3athrow commented:
Closed by be4f5e76f25a75ed09cff693d9c20a7a5e0a8f6f
